File: app/src/main/java/com/mancode/financetracker/database/pojos/BalanceExtended.java
package com.mancode.financetracker.database.pojos;

import com.mancode.financetracker.database.converter.DateConverter;

import org.threeten.bp.LocalDate;

import androidx.room.ColumnInfo;
import androidx.room.TypeConverters;

/**
 * Created by Manveru on 03.02.2018.
 */

public class BalanceExtended {

    @ColumnInfo(name = "_id")
    public int id;

    @ColumnInfo(name = "balance_check_date")
    @TypeConverters(DateConverter.class)
    public LocalDate checkDate;

    @ColumnInfo(name = "balance_value")
    public double value;

    @ColumnInfo(name = "account_name")
    public String accountName;

    @ColumnInfo(name = "account_type")
    public int accountType;

    @ColumnInfo(name = "account_currency")
    public String accountCurrency;

    public BalanceExtended(int id, LocalDate checkDate,
                           double value, String accountName,
                           int accountType, String accountCurrency) {
        this.id = id;
        this.checkDate = checkDate;
        this.value = value;
        this.accountName = accountName;
        this.accountType = accountType;
        this.accountCurrency = accountCurrency;
    }
}

File: app/src/main/java/com/mancode/financetracker/database/pojos/BalanceExtended.kt
package com.mancode.financetracker.database.pojos

import androidx.room.ColumnInfo
import androidx.room.TypeConverters
import com.mancode.financetracker.database.converter.DateConverter
import org.threeten.bp.LocalDate

/**
 * Created by Manveru on 03.02.2018.
 */

class BalanceExtended(@field:ColumnInfo(name = "_id")
                      var id: Int, @field:ColumnInfo(name = "balance_check_date")
                      @field:TypeConverters(DateConverter::class)
                      var checkDate: LocalDate,
                      @field:ColumnInfo(name = "balance_value")
                      var value: Double, @field:ColumnInfo(name = "account_name")
                      var accountName: String,
                      @field:ColumnInfo(name = "account_type")
                      var accountType: Int, @field:ColumnInfo(name = "account_currency")
                      var accountCurrency: String)
